src/api.rs

```diff
@@ -0,0 +1,90 @@
+//! This file contains the API for the runtime GC. Interaction with the GC is
+//! only legal through these functions. 
+//!
+//! The GC runtime consists of a mutable singleton `Collector` struct: 
+//!
+//! ```rust, ignore
+//! struct Collector { 
+//!     // Heap bookkeeping information
+//!     hstart: *const usize,
+//!     hptr: *mut usize,
+//!
+//!     // Flag to determine whether to collect at the next safepoint
+//!     collect_next: bool,
+//!
+//!     // The in-memory safepoint table used to identify roots
+//!     roots: HashMap<ReturnAddress, SafepointRoots>
+//! }
+
+use collector::COLLECTOR;
+
+/// This must be called before the GC can be used (usually in the setup code
+/// before `main()`). Initialisation consists of two stages:
+///     1. Read the stackmap section in the ELF file into an in-memory table for
+///        fast lookup.
+///     2. Allocate a chunk of heap memory to be used to store objects managed
+///        by the GC.
+pub fn init() {
+        COLLECTOR.alloc_heap().make_root_table();
```

Comment:
I think this file hasn't gone through `rustfmt` ;)

Reply:
Guilty